client/packages/invoices/src/InboundShipment/api/hooks/line/useZeroInboundLinesQuantity.ts

```diff
@@ -0,0 +1,52 @@
+import {
+  useTableStore,
+  useTranslation,
+  useDeleteConfirmation,
+} from '@openmsupply-client/common';
+import { useIsInboundDisabled } from '../utils/useIsInboundDisabled';
+import { useInboundRows } from './useInboundRows';
+import { useZeroInboundLineQuantity } from './useZeroInboundLineQuantity';
+
+export const useZeroInboundLinesQuantity = (): (() => void) => {
+  const { items, lines } = useInboundRows();
+  const { mutateAsync } = useZeroInboundLineQuantity();
+  const isDisabled = useIsInboundDisabled();
+  const t = useTranslation('replenishment');
+
+  const selectedRows =
+    useTableStore(state => {
+      const { isGrouped } = state;
+
+      return isGrouped
+        ? items
+            ?.filter(({ id }) => state.rowState[id]?.isSelected)
+            .map(({ lines }) => lines.flat())
+            .flat()
+        : lines?.filter(({ id }) => state.rowState[id]?.isSelected);
+    }) || [];
+
+  const notAllLinesSelected = selectedRows.length !== lines?.length;
+
+  const onZeroQuantities = async () => {
+    await mutateAsync(selectedRows).catch(err => {
+      throw err;
+    });
+  };
+
+  const confirmAndZeroLines = useDeleteConfirmation({
+    selectedRows,
+    deleteAction: onZeroQuantities,
+    canDelete: !isDisabled && !notAllLinesSelected,
```

Comment:
Not sure if user has to select all lines to 0 or not... but seemed to make sense to me since the invoice gets finalised 👀

Reply:
Where does the invoice gets finalized? does the user has to do it manually, or am I missing it in the code?

Reply:
Oh this was changed sorry forgot to delete